Lecture2/Task2/com/playtika/Student.java

```diff
@@ -0,0 +1,150 @@
+
+import static java.lang.System.out;
+
+public class Student {
+    private int id;
+    private String name;
+    private String surname;
+    private int age;
+
+    public Student(int id, String name, String surname, int age) {
+        this.id = id;
+        this.name = name;
+        this.surname = surname;
+        this.age = age;
+    }
+
+    public int getAge() {
+
+        return age;
+    }
+
+    public String getSurname() {
+
+        return surname;
+    }
+
+    public int getId() {
+
+        return id;
+    }
+
+    public static void main(String[] args) {
+
+        List<Student> studentList = new ArrayList<>();
+        Student firstStudent = new Student(1, "Vova", "Ivanov", 18);
+        studentList.add(firstStudent);
+        Student secondStudent = new Student(2, "Lesha", "Lebedev", 19);
+        studentList.add(secondStudent);
+        Student thirdStudent = new Student(3, "Vanya", "Petrov", 21);
+        studentList.add(thirdStudent);
+        Student fourthStudent = new Student(4, "Vanya", "Ivanov", 20);
+        studentList.add(fourthStudent);
+        Student fifthStudent = new Student(5, "Sergey", "Sidorov", 22);
+        studentList.add(fifthStudent);
+        Student sixthStudent = new Student(6, "Andrey", "Ivanov", 20);
+        studentList.add(sixthStudent);
+        Student seventhStudent = new Student(7, "Igor", "Makarevich", 18);
+        studentList.add(seventhStudent);
+        Student eighthStudent = new Student(8, "Igor", "Nazarov", 19);
+        studentList.add(eighthStudent);
+        Student ninthStudent = new Student(9, "Artem", "Petrov", 24);
+        studentList.add(ninthStudent);
+        Student tenthStudent = new Student(10, "Pavel", "Nazarov", 21);
+        studentList.add(tenthStudent);
+        for (Student st : studentList) {
+            out.println(st.toString());
+        }
+        Collections.sort(studentList, new AgeComparator());
+        out.println(studentList);
+
+        surnameStartsWithI(studentList);
+        getAverageAge(studentList);
+        HashMap<Integer, Student> integerStudentHashMap = convertToMap(studentList);
+        Map<Integer, Student> integerStudentMap = selectedByKey(integerStudentHashMap);
+
+    }
+
+    @Override
+    public String toString() {
+        return "Student{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                ", surname='" + surname + '\'' +
+                ", age=" + age +
+                '}';
+    }
+
+    public static HashMap<Integer, Student> convertToMap(List<Student> studentList) {
+        HashMap<Integer, Student> studentsHM = new HashMap<>(); //convert list to map
+
+        for (Student student1 : studentList) {
+            studentsHM.put((student1.getId()), student1);
+        }
+        System.out.println("Student's HashMap is: " + studentsHM);
+        return studentsHM;
+
+    }
+
+    private static Map<Integer, Student> selectedByKey(HashMap<Integer, Student> studentsHM) {
```

Comment:
Is it really required for `studentsHM ` to be a HashMap?

Reply:
According task  "method returns Map"